### PR TITLE
Merging in components to run on Mac OS

### PR DESCRIPTION
**Pull request by ChadBrow**
This should have no effect on running the application on Linux.

### CODE REVIEW

**Review by BCaven — CHANGES_REQUESTED** [review]
this will need to be added back in so the development server can build the vue site

File: src/macOS_pyaudio.py
Comment (BCaven) [inline src/macOS_pyaudio.py:1]: eventually we will need to support switching input devices and settings
also, this uses the 0.0.0.0 formatting which will probably not be supported in the final version - but we will have to see, mac might still interpret it correctly even if the server will not recognize 0.0.0.0

File: src/package-lock.json
Comment (BCaven) [inline src/package-lock.json:1]: wait, why is there a `package-lock` in the `src/` folder?

**Review by BCaven — DISMISSED** [review]
answer the questions sir

**Comment by BCaven** [comment]
After changing the dockerfile back to building the vue site, this is compatible with linux.
Why was it removed from the mac build again?

It only really matters for development, because production versions will pre-build the vue site

**Review by AnarBayanzul — APPROVED** [review]
looks good to me!

**Pull request merged**